morphia/src/main/java/org/mongodb/morphia/converters/BigDecimalConverter.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2016 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mongodb.morphia.converters;
+
+import org.bson.types.Decimal128;
+import org.mongodb.morphia.mapping.MappedField;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+/**
+ * This provides a convert to/from BigDecimal in applications using the Decimal128 type introduced in the 3.4.0 release of MongoDB.
+ *
+ * @mongodb.server.release 3.4
+ * @see Decimal128
+ * @since 1.3
+ */
+public class BigDecimalConverter extends TypeConverter implements SimpleValueConverter {
+    @Override
+    public Object decode(final Class<?> targetClass, final Object value, final MappedField optionalExtraInfo) {
+        if (value == null) {
+            return null;
+        }
+
+        if (value instanceof BigDecimal) {
```

Comment:
First time I've looked carefully at a TypeConverter subclass, so for my edification can you explain the circumstances under which you'd expect to be decoding anything but a Decimal128 instance?